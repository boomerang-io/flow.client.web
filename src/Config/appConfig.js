// Look for the data injected into the HTML file from the Express app
// See server/app.js for implementation
export const APP_ROOT =
  window._SERVER_DATA && window._SERVER_DATA.APP_ROOT ? window._SERVER_DATA.APP_ROOT : "/BMRG_APP_ROOT_CONTEXT";

export const PRODUCT_STANDALONE =
  window._SERVER_DATA && window._SERVER_DATA.PRODUCT_STANDALONE
    ? window._SERVER_DATA.PRODUCT_STANDALONE === "true"
    : process.env.REACT_APP_PRODUCT_STANDALONE
    ? true
    : false;

export const EMBEDDED_MODE =
  window._SERVER_DATA && window._SERVER_DATA.EMBEDDED_MODE
    ? window._SERVER_DATA.EMBEDDED_MODE === "true"
    : process.env.REACT_APP_EMBEDDED_MODE
    ? true
    : true;

export const CORE_ENV_URL =
  window._SERVER_DATA && window._SERVER_DATA.CORE_ENV_URL ? window._SERVER_DATA.CORE_ENV_URL : "";

export const BASE_DOCUMENTATION_URL = PRODUCT_STANDALONE
  ? "https://www.useboomerang.io/docs/boomerang-flow"
  : `${CORE_ENV_URL}/docs/boomerang-flow`;

export const isDevEnv = process.env.NODE_ENV === "development";
export const isTestEnv = process.env.NODE_ENV === "test";

export const AppPath = {
  Root: "/",
  Error: "/error",
  Activity: "/activity",
  Editor: "/editor/:workflowId",
  EditorDesigner: `/editor/:workflowId/workflow`,
  EditorConfigure: `editor/:workflowId/configure`,
  EditorChangelog: `/editor/:workflowId/changelog`,
  EditorProperties: `/editor/:workflowId/parameters`,
  Execution: "/activity/:workflowId/execution/:executionId",
  Insights: "/insights",
  Properties: "/admin/parameters",
  Quotas: "/admin/quotas",
  QuotasEdit: "/admin/quotas/:teamId",
  Settings: "/admin/settings",
  SystemWorkflows: "/admin/system-workflows",
  TaskTemplates: "/admin/task-templates",
  TaskTemplateEdit: `/admin/task-templates/:id/:version`,
  TaskTemplateYaml: `/admin/task-templates/:id/:version/yaml-editor`,
  Team: "/admin/teams/:teamId",
  TeamSettings: "/admin/teams/:teamId/settings",
  TeamWorkflows: "/admin/teams/:teamId/workflows",
  TeamList: "/admin/teams",
  TeamProperties: `/manage/team-parameters`,
  User: "/admin/users/:userId",
  UserList: "/admin/users",

  Workflows: "/workflows",
  WorkflowsMine: "/workflows/mine",
  WorkflowsTeams: "/workflows/teams",

  ManageTaskTemplates: `/manage/task-templates`,
  ManageTaskTemplatesTeam: `/manage/task-templates/team/:teamId`,
  ManageTaskTemplateEdit: `/manage/task-templates/team/:teamId/:taskId/:version`,
  ManageTaskTemplateYaml: `/manage/task-templates/team/:teamId/:taskId/:version/yaml-editor`,
};

export const appLink = {
  activity: () => `/activity`,
  editorDesigner: ({ workflowId }) => `/editor/${workflowId}/workflow`,
  editorConfigure: ({ workflowId }) => `/editor/${workflowId}/configure`,
  editorChangelog: ({ workflowId }) => `/editor/${workflowId}/changelog`,
  editorProperties: ({ workflowId }) => `/editor/${workflowId}/parameters`,
  execution: ({ executionId, workflowId }) => `/activity/${workflowId}/execution/${executionId}`,
  insights: () => "/insights",
  properties: () => "/admin/parameters",
  manageUsers: () => "/admin/users",
  quotas: () => "/admin/quotas",
  quotasEdit: ({ teamId }) => `/admin/quotas/${teamId}`,
  settings: () => "/admin/settings",
  systemWorkflows: () => "/admin/system-workflows",
  taskTemplates: () => "/admin/task-templates",
  taskTemplateEdit: ({ id, version }) => `/admin/task-templates/${id}/${version}`,
  taskTemplateYaml: ({ id, version }) => `/admin/task-templates/${id}/${version}/yaml-editor`,
  teamProperties: () => `/manage/team-parameters`,
  teamTaskTemplates: () => `/manage/task-templates`,
  team: ({ teamId }) => `/admin/teams/${teamId}`,
  teamWorkflows: ({ teamId }) => `/admin/teams/${teamId}/workflows`,
  teamSettings: ({ teamId }) => `/admin/teams/${teamId}/settings`,
  teamList: () => "/admin/teams",
  workflows: () => "/workflows",
  workflowsMine: () => "/workflows/mine",
  workflowsTeams: () => "/workflows/teams",
  workflowActivity: ({ workflowId }) => `/activity?page=0&size=10&workflowIds=${workflowId}`,

  manageTaskTemplates: ({ teamId }) => `/manage/task-templates/team/${teamId}`,
  manageTaskTemplateEdit: ({ teamId, taskId, version }) => `/manage/task-templates/team/${teamId}/${taskId}/${version}`,
  manageTaskTemplateYaml: ({ teamId, taskId, version }) =>
    `/manage/task-templates/team/${teamId}/${taskId}/${version}/yaml-editor`,

  //external apps
  docsWorkflowEditor: () => `${BASE_DOCUMENTATION_URL}/how-to-guide/workflow-editor`,
};

export const queryStringOptions = { arrayFormat: "comma", skipEmptyString: true };

export const FeatureFlag = {
  /**
   * new Feature Flags
   */
  TeamManagementEnabled: "TeamManagementEnabled",
  WorkflowQuotasEnabled: "WorkflowQuotasEnabled",
  SettingsEnabled: "SettingsEnabled",
  UserManagementEnabled: "UserManagementEnabled",
  GlobalParametersEnabled: "GlobalParametersEnabled",
  WorkflowTokensEnabled: "WorkflowTokensEnabled",
  TaskManagerEnabled: "TaskManagerEnabled",
  EditVerifiedTasksEnabled: "EditVerifiedTasksEnabled",
  WorkflowTriggersEnabled: "WorkflowTriggersEnabled",
  TeamParametersEnabled: "TeamParametersEnabled",

  ActivityEnabled: "ActivityEnabled",
  InsightsEnabled: "InsightsEnabled",
};
