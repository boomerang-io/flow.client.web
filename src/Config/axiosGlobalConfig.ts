// Setup global config for Axios requests so they work with ICP
import axios from "axios";
axios.defaults.withCredentials = true;
