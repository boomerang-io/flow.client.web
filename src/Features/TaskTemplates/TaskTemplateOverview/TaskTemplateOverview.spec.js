import React from "react";
import { Route } from "react-router-dom";
import TaskTemplateOverview from "./index";
import { startApiServer } from "ApiServer";
import { AppPath, appLink } from "Config/appConfig";
import { tasktemplate } from "ApiServer/fixtures";

let server;

beforeEach(() => {
  server = startApiServer();
});

afterEach(() => {
  server.shutdown();
});

const props = {
  taskTemplates: tasktemplate,
  updateTemplateInState: jest.fn(),
  editVerifiedTasksEnabled: true,
};

describe("TaskTemplateOverview --- Snapshot", () => {
  it("Capturing Snapshot of Task Templates", async () => {
    const { baseElement, findByText } = rtlContextRouterRender(
      <Route path={AppPath.TaskTemplateEdit}>
        <TaskTemplateOverview {...props} />
      </Route>,
      { route: appLink.taskTemplateEdit({id: "5e670a1e2d5e6a302de4f41d", version: "1"}) }
    );
    await findByText(/Build the definition requirements for this task/i);
    expect(baseElement).toMatchSnapshot();
  });
});
