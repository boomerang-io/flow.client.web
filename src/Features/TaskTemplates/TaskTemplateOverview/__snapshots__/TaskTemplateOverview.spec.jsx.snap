// Vitest Snapshot v1

exports[`TaskTemplateOverview --- Snapshot > Capturing Snapshot of Task Templates 1`] = `
<body
  id="app"
>
  <div>
    <div
      class="_container_734922"
    >
      <header
        class="cds--bmrg-feature-header _featureHeader_4b0ee6 --bordered"
      >
        <section
          class="cds--bmrg-feature-header__content"
        >
          <p
            class="_category_4b0ee6"
          >
            communication
          </p>
          <div
            class="_infoContainer_4b0ee6"
          >
            <svg
              alt="Task node type get, read"
              aria-hidden="true"
              fill="currentColor"
              focusable="false"
              height="16"
              preserveAspectRatio="xMidYMid meet"
              style="min-width: 1.5rem; min-height: 1.5rem; margin-right: 0.75rem;"
              viewBox="0 0 16 16"
              width="16"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M15,14.3L10.7,10c1.9-2.3,1.6-5.8-0.7-7.7S4.2,0.7,2.3,3S0.7,8.8,3,10.7c2,1.7,5,1.7,7,0l4.3,4.3L15,14.3z M2,6.5 C2,4,4,2,6.5,2S11,4,11,6.5S9,11,6.5,11S2,9,2,6.5z"
              />
            </svg>
            <h1
              class="_taskName_4b0ee6"
              title="Slack User Look Up"
            >
              Slack User Look Up
            </h1>
            <button
              class="_button_61c4b8"
            >
              <svg
                aria-hidden="true"
                fill="currentColor"
                focusable="false"
                height="16"
                preserveAspectRatio="xMidYMid meet"
                viewBox="0 0 32 32"
                width="16"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M26,2H8A2,2,0,0,0,6,4V8H4v2H6v5H4v2H6v5H4v2H6v4a2,2,0,0,0,2,2H26a2,2,0,0,0,2-2V4A2,2,0,0,0,26,2Zm0,26H8V24h2V22H8V17h2V15H8V10h2V8H8V4H26Z"
                />
                <path
                  d="M14 8H22V10H14zM14 15H22V17H14zM14 22H22V24H14z"
                />
              </svg>
            </button>
            <a
              class="_exportYaml_4b0ee6"
              download="Slack User Look Up.yaml"
              href="/api/workflow/tasktemplate/5e670a1e2d5e6a302de4f41d/yaml/1"
              rel="noopener noreferrer"
              target="_blank"
            >
              <svg
                aria-hidden="true"
                fill="currentColor"
                focusable="false"
                height="16"
                preserveAspectRatio="xMidYMid meet"
                viewBox="0 0 16 16"
                width="16"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M13 7L12.3 6.3 8.5 10.1 8.5 1 7.5 1 7.5 10.1 3.7 6.3 3 7 8 12zM13 12v2H3v-2H2v2l0 0c0 .6.4 1 1 1h10c.6 0 1-.4 1-1l0 0v-2H13z"
                />
              </svg>
            </a>
          </div>
          <p
            class="_lastUpdate_4b0ee6"
          >
            Version created Apr 30, 2020 by ---
          </p>
          <div
            class="cds--bmrg-feature-header__footer"
          >
            <nav
              aria-label="Task template views"
              class="cds--tabs"
            >
              <ul
                class="cds--tab--list"
              >
                <a
                  aria-current="page"
                  class="cds--tabs__nav-item cds--tabs__nav-link cds--tabs__nav-item--selected"
                  href="/admin/task-templates/5e670a1e2d5e6a302de4f41d/1"
                >
                  Overview
                </a>
                <a
                  class="cds--tabs__nav-item cds--tabs__nav-link"
                  href="/admin/task-templates/5e670a1e2d5e6a302de4f41d/1/yaml-editor"
                >
                  YAML
                </a>
              </ul>
            </nav>
          </div>
        </section>
        <div
          class="_buttons_4b0ee6"
        >
          <div
            class="_container_b033d1"
          >
            <div
              class="_buttonListContainer_b033d1"
            >
              <div
                class="_buttonList_b033d1"
              >
                <button
                  class="_button_b033d1"
                  disabled=""
                >
                  <svg
                    alt="first version"
                    aria-hidden="true"
                    class="_icon_b033d1 _disabled_b033d1"
                    fill="currentColor"
                    focusable="false"
                    height="16"
                    preserveAspectRatio="xMidYMid meet"
                    viewBox="0 0 32 32"
                    width="16"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M14 16L24 6 25.4 7.4 16.8 16 25.4 24.6 24 26zM8 4H10V28H8z"
                    />
                  </svg>
                </button>
                <button
                  class="_button_b033d1"
                  disabled=""
                >
                  <svg
                    alt="back one version"
                    aria-hidden="true"
                    class="_icon_b033d1 _disabled_b033d1"
                    fill="currentColor"
                    focusable="false"
                    height="16"
                    preserveAspectRatio="xMidYMid meet"
                    viewBox="0 0 16 16"
                    width="16"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M5 8L10 3 10.7 3.7 6.4 8 10.7 12.3 10 13z"
                    />
                  </svg>
                </button>
              </div>
              <p
                class="_versionText_b033d1"
              >
                Version 1
              </p>
              <div
                class="_buttonList_b033d1"
              >
                <button
                  class="_button_b033d1"
                  disabled=""
                >
                  <svg
                    alt="forward one version"
                    aria-hidden="true"
                    class="_icon_b033d1 _disabled_b033d1"
                    fill="currentColor"
                    focusable="false"
                    height="16"
                    preserveAspectRatio="xMidYMid meet"
                    viewBox="0 0 16 16"
                    width="16"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M11 8L6 13 5.3 12.3 9.6 8 5.3 3.7 6 3z"
                    />
                  </svg>
                </button>
                <button
                  class="_button_b033d1"
                  disabled=""
                >
                  <svg
                    alt="last version"
                    aria-hidden="true"
                    class="_icon_b033d1 _disabled_b033d1"
                    fill="currentColor"
                    focusable="false"
                    height="16"
                    preserveAspectRatio="xMidYMid meet"
                    viewBox="0 0 32 32"
                    width="16"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M18 16L8 26 6.6 24.6 15.2 16 6.6 7.4 8 6zM22 4H24V28H22z"
                    />
                  </svg>
                </button>
              </div>
            </div>
          </div>
          <button
            class="_resetButton_4b0ee6 cds--btn cds--btn--md cds--btn--ghost cds--btn--disabled"
            disabled=""
            tabindex="0"
            type="button"
          >
             
            Reset changes
            <svg
              aria-hidden="true"
              class="cds--btn__icon"
              fill="currentColor"
              focusable="false"
              height="16"
              preserveAspectRatio="xMidYMid meet"
              viewBox="0 0 16 16"
              width="16"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M12,8c1.2,0,2,0.8,2,2c0,1.2-0.8,2-2,2H9v1h3c1.7,0,3-1.2,3-3c0-1.7-1.3-3-3-3L2.9,7l2-2L4.2,4.3L1,7.5 l3.2,3.2L4.9,10l-2-2L12,8z"
              />
            </svg>
          </button>
          <button
            class="_mainActionButton_4b0ee6 cds--btn cds--btn--md cds--btn--primary cds--btn--disabled"
            disabled=""
            tabindex="0"
            type="button"
          >
            Save...
            <svg
              aria-hidden="true"
              aria-label="Save a new version or update the current one"
              class="cds--btn__icon"
              fill="currentColor"
              focusable="false"
              height="16"
              preserveAspectRatio="xMidYMid meet"
              role="img"
              viewBox="0 0 16 16"
              width="16"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M13.9,4.6l-2.5-2.5C11.3,2.1,11.1,2,11,2H3C2.4,2,2,2.4,2,3v10c0,0.6,0.4,1,1,1h10c0.6,0,1-0.4,1-1V5 C14,4.9,13.9,4.7,13.9,4.6z M6,3h4v2H6V3z M10,13H6V9h4V13z M11,13V9c0-0.6-0.4-1-1-1H6C5.4,8,5,8.4,5,9v4H3V3h2v2c0,0.6,0.4,1,1,1 h4c0.6,0,1-0.4,1-1V3.2l2,2V13H11z"
              />
            </svg>
          </button>
        </div>
      </header>
      <div
        class="_content_734922"
      >
        <section
          class="_taskActionsSection_734922"
        >
          <p
            class="_description_734922"
          >
            Build the definition requirements for this task.
          </p>
          <button
            class="_archive_734922 cds--btn cds--btn--md cds--btn--ghost"
            tabindex="0"
            type="button"
          >
            Archive
            <svg
              aria-hidden="true"
              aria-label="Archive"
              class="cds--btn__icon"
              fill="currentColor"
              focusable="false"
              height="16"
              preserveAspectRatio="xMidYMid meet"
              role="img"
              viewBox="0 0 32 32"
              width="16"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M14 19H18V21H14z"
              />
              <path
                d="M6,2V28a2,2,0,0,0,2,2H24a2,2,0,0,0,2-2V2ZM24,28H8V16H24Zm0-14H8V10H24ZM8,8V4H24V8Z"
              />
            </svg>
          </button>
        </section>
        <div
          class="_detailCardsContainer_734922"
        >
          <div
            class="cds--tile _editDetailsCard_734922"
          >
            <section
              class="_editTitle_734922"
            >
              <h1>
                Basics
              </h1>
              <button
                class="cds--btn cds--btn--md cds--btn--ghost"
                tabindex="0"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  aria-label="edit-template"
                  class="cds--btn__icon"
                  fill="currentColor"
                  focusable="false"
                  height="16"
                  preserveAspectRatio="xMidYMid meet"
                  role="img"
                  viewBox="0 0 32 32"
                  width="16"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M2 26H30V28H2zM25.4 9c.8-.8.8-2 0-2.8 0 0 0 0 0 0l-3.6-3.6c-.8-.8-2-.8-2.8 0 0 0 0 0 0 0l-15 15V24h6.4L25.4 9zM20.4 4L24 7.6l-3 3L17.4 7 20.4 4zM6 22v-3.6l10-10 3.6 3.6-10 10H6z"
                  />
                </svg>
              </button>
            </section>
            <dl
              class="_detailsDataList_734922"
            >
              <section
                class="_infoSection_734922"
              >
                <dt
                  class="_label_734922"
                >
                  Name
                </dt>
                <dd
                  class="_value_734922"
                  data-testid="Name"
                >
                  Slack User Look Up
                </dd>
              </section>
              <section
                class="_infoSection_734922"
              >
                <dt
                  class="_label_734922"
                >
                  Category
                </dt>
                <dd
                  class="_value_734922"
                  data-testid="Category"
                >
                  communication
                </dd>
              </section>
              <section
                class="_infoSection_734922"
              >
                <dt
                  class="_label_734922"
                >
                  Icon
                </dt>
                <div
                  class="_basicIcon_734922"
                >
                  <svg
                    alt="Task node type get, read"
                    aria-hidden="true"
                    fill="currentColor"
                    focusable="false"
                    height="16"
                    preserveAspectRatio="xMidYMid meet"
                    style="width: 1.5rem; height: 1.5rem; margin-right: 0.75rem;"
                    viewBox="0 0 16 16"
                    width="16"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M15,14.3L10.7,10c1.9-2.3,1.6-5.8-0.7-7.7S4.2,0.7,2.3,3S0.7,8.8,3,10.7c2,1.7,5,1.7,7,0l4.3,4.3L15,14.3z M2,6.5 C2,4,4,2,6.5,2S11,4,11,6.5S9,11,6.5,11S2,9,2,6.5z"
                    />
                  </svg>
                  <p
                    class="_value_734922"
                  >
                    Search
                  </p>
                </div>
              </section>
              <section
                class="_infoSection_734922"
              >
                <dt
                  class="_label_734922"
                >
                  Description
                </dt>
                <dd
                  class="_value_734922"
                  data-testid="Description"
                >
                  Look up Slack user by email address
                </dd>
              </section>
              <section
                class="_infoSection_734922"
              >
                <dt
                  class="_label_734922"
                >
                  Image
                </dt>
                <dd
                  class="_noValue_734922"
                  data-testid="Image"
                >
                  Not defined yet
                </dd>
              </section>
              <section
                class="_infoSection_734922"
              >
                <dt
                  class="_label_734922"
                >
                  Command
                </dt>
                <dd
                  class="_value_734922"
                  data-testid="Command"
                >
                  npm start
run test
cd ..
                </dd>
              </section>
              <section
                class="_infoSection_734922"
              >
                <dt
                  class="_label_734922"
                >
                  Arguments
                </dt>
                <dd
                  class="_value_734922"
                  data-testid="Arguments"
                >
                  slack
lookUpUser
                </dd>
              </section>
              <section
                class="_infoSection_734922"
              >
                <dt
                  class="_label_734922"
                >
                  Script
                </dt>
                <dd
                  class="_noValue_734922"
                  data-testid="Script"
                >
                  Not defined yet
                </dd>
              </section>
              <section
                class="_infoSection_734922"
              >
                <dt
                  class="_label_734922"
                >
                  Working Directory
                </dt>
                <dd
                  class="_noValue_734922"
                  data-testid="Working Directory"
                >
                  Not defined yet
                </dd>
              </section>
              <section
                class="_infoSection_734922"
              >
                <dt
                  class="_label_734922"
                >
                  Envs
                </dt>
                <dd
                  class="_noValue_734922"
                  data-testid="Envs"
                >
                  Not defined yet
                </dd>
              </section>
              <section
                class="_infoSection_734922"
              >
                <dt
                  class="_label_734922"
                >
                  Contribution level
                </dt>
                <div
                  class="_basicIcon_734922"
                >
                  <p
                    class="_value_734922"
                  >
                    Verified task
                  </p>
                  <svg
                    aria-hidden="true"
                    fill="#0072C3"
                    focusable="false"
                    height="16"
                    preserveAspectRatio="xMidYMid meet"
                    style="margin-left: 0.5rem;"
                    viewBox="0 0 32 32"
                    width="16"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M16,2a9,9,0,0,0-6,15.69V30l6-4,6,4V17.69A9,9,0,0,0,16,2Zm4,24.26-2.89-1.92L16,23.6l-1.11.74L12,26.26V19.05a8.88,8.88,0,0,0,8,0ZM20.89,16A7,7,0,1,1,23,11,7,7,0,0,1,20.89,16Z"
                    />
                  </svg>
                </div>
              </section>
            </dl>
          </div>
          <div
            class="cds--tile _editFieldsCard_734922"
          >
            <section
              class="_editTitle_734922"
            >
              <hgroup
                class="_fieldsTitle_734922"
              >
                <h1>
                  Definition fields
                </h1>
                <h2
                  class="_fieldDesc_734922"
                >
                  Drag to reorder the fields
                </h2>
              </hgroup>
              <div
                class="_fieldActions_734922"
              >
                <button
                  class="cds--btn cds--btn--md cds--btn--ghost"
                  style="width: 7rem;"
                  tabindex="0"
                  type="button"
                >
                  Preview
                  <svg
                    aria-hidden="true"
                    aria-label="Preview task"
                    class="cds--btn__icon"
                    fill="currentColor"
                    focusable="false"
                    height="16"
                    preserveAspectRatio="xMidYMid meet"
                    role="img"
                    viewBox="0 0 16 16"
                    width="16"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M15.5,7.8C14.3,4.7,11.3,2.6,8,2.5C4.7,2.6,1.7,4.7,0.5,7.8c0,0.1,0,0.2,0,0.3c1.2,3.1,4.1,5.2,7.5,5.3 c3.3-0.1,6.3-2.2,7.5-5.3C15.5,8.1,15.5,7.9,15.5,7.8z M8,12.5c-2.7,0-5.4-2-6.5-4.5c1-2.5,3.8-4.5,6.5-4.5s5.4,2,6.5,4.5 C13.4,10.5,10.6,12.5,8,12.5z"
                    />
                    <path
                      d="M8,5C6.3,5,5,6.3,5,8s1.3,3,3,3s3-1.3,3-3S9.7,5,8,5z M8,10c-1.1,0-2-0.9-2-2s0.9-2,2-2s2,0.9,2,2S9.1,10,8,10z"
                    />
                  </svg>
                </button>
                <button
                  class="cds--btn cds--btn--md cds--btn--ghost"
                  style="width: 8rem;"
                  tabindex="0"
                  type="button"
                >
                  Add a field
                  <svg
                    aria-hidden="true"
                    aria-label="Add field"
                    class="cds--btn__icon"
                    fill="currentColor"
                    focusable="false"
                    height="16"
                    preserveAspectRatio="xMidYMid meet"
                    role="img"
                    viewBox="0 0 32 32"
                    width="16"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M17 15L17 8 15 8 15 15 8 15 8 17 15 17 15 24 17 24 17 17 24 17 24 15z"
                    />
                  </svg>
                </button>
              </div>
            </section>
            <div
              class="_fieldsContainer_734922"
            >
              <section
                class="_fieldSection_734922"
                data-rbd-draggable-context-id="0"
                data-rbd-draggable-id="token"
              >
                <div
                  aria-describedby="rbd-hidden-text-0-hidden-text-0"
                  class="_iconContainer_734922"
                  data-rbd-drag-handle-context-id="0"
                  data-rbd-drag-handle-draggable-id="token"
                  draggable="false"
                  role="button"
                  style="display: flex;"
                  tabindex="0"
                >
                  <svg
                    aria-hidden="true"
                    class="_dragabble_734922"
                    fill="currentColor"
                    focusable="false"
                    height="16"
                    preserveAspectRatio="xMidYMid meet"
                    viewBox="0 0 32 32"
                    width="16"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M10 6H14V10H10zM18 6H22V10H18zM10 14H14V18H10zM18 14H22V18H18zM10 22H14V26H10zM18 22H22V26H18z"
                    />
                  </svg>
                </div>
                <dd
                  class="_value_734922"
                  data-testid="Token"
                  style="margin-left: 0px;"
                >
                  Text | Token - token
                </dd>
                <div
                  class="_actions_734922"
                >
                  <button
                    class="cds--btn cds--btn--md cds--btn--ghost"
                    tabindex="0"
                    type="button"
                  >
                    <svg
                      aria-hidden="true"
                      aria-label="Edit field"
                      class="cds--btn__icon"
                      fill="currentColor"
                      focusable="false"
                      height="16"
                      preserveAspectRatio="xMidYMid meet"
                      role="img"
                      viewBox="0 0 32 32"
                      width="16"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M2 26H30V28H2zM25.4 9c.8-.8.8-2 0-2.8 0 0 0 0 0 0l-3.6-3.6c-.8-.8-2-.8-2.8 0 0 0 0 0 0 0l-15 15V24h6.4L25.4 9zM20.4 4L24 7.6l-3 3L17.4 7 20.4 4zM6 22v-3.6l10-10 3.6 3.6-10 10H6z"
                      />
                    </svg>
                  </button>
                  <button
                    class="_delete_734922 cds--btn cds--btn--md cds--btn--ghost"
                    tabindex="0"
                    type="button"
                  >
                    <svg
                      aria-hidden="true"
                      aria-label="delete-field"
                      class="cds--btn__icon"
                      fill="currentColor"
                      focusable="false"
                      height="16"
                      preserveAspectRatio="xMidYMid meet"
                      role="img"
                      viewBox="0 0 32 32"
                      width="16"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M12 12H14V24H12zM18 12H20V24H18z"
                      />
                      <path
                        d="M4 6V8H6V28a2 2 0 002 2H24a2 2 0 002-2V8h2V6zM8 28V8H24V28zM12 2H20V4H12z"
                      />
                    </svg>
                  </button>
                </div>
              </section>
              <section
                class="_fieldSection_734922"
                data-rbd-draggable-context-id="0"
                data-rbd-draggable-id="emailAddress"
              >
                <div
                  aria-describedby="rbd-hidden-text-0-hidden-text-0"
                  class="_iconContainer_734922"
                  data-rbd-drag-handle-context-id="0"
                  data-rbd-drag-handle-draggable-id="emailAddress"
                  draggable="false"
                  role="button"
                  style="display: flex;"
                  tabindex="0"
                >
                  <svg
                    aria-hidden="true"
                    class="_dragabble_734922"
                    fill="currentColor"
                    focusable="false"
                    height="16"
                    preserveAspectRatio="xMidYMid meet"
                    viewBox="0 0 32 32"
                    width="16"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M10 6H14V10H10zM18 6H22V10H18zM10 14H14V18H10zM18 14H22V18H18zM10 22H14V26H10zM18 22H22V26H18z"
                    />
                  </svg>
                </div>
                <dd
                  class="_value_734922"
                  data-testid="Email Address"
                  style="margin-left: 0px;"
                >
                  Text | Email Address - emailAddress
                </dd>
                <div
                  class="_actions_734922"
                >
                  <button
                    class="cds--btn cds--btn--md cds--btn--ghost"
                    tabindex="0"
                    type="button"
                  >
                    <svg
                      aria-hidden="true"
                      aria-label="Edit field"
                      class="cds--btn__icon"
                      fill="currentColor"
                      focusable="false"
                      height="16"
                      preserveAspectRatio="xMidYMid meet"
                      role="img"
                      viewBox="0 0 32 32"
                      width="16"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M2 26H30V28H2zM25.4 9c.8-.8.8-2 0-2.8 0 0 0 0 0 0l-3.6-3.6c-.8-.8-2-.8-2.8 0 0 0 0 0 0 0l-15 15V24h6.4L25.4 9zM20.4 4L24 7.6l-3 3L17.4 7 20.4 4zM6 22v-3.6l10-10 3.6 3.6-10 10H6z"
                      />
                    </svg>
                  </button>
                  <button
                    class="_delete_734922 cds--btn cds--btn--md cds--btn--ghost"
                    tabindex="0"
                    type="button"
                  >
                    <svg
                      aria-hidden="true"
                      aria-label="delete-field"
                      class="cds--btn__icon"
                      fill="currentColor"
                      focusable="false"
                      height="16"
                      preserveAspectRatio="xMidYMid meet"
                      role="img"
                      viewBox="0 0 32 32"
                      width="16"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M12 12H14V24H12zM18 12H20V24H18z"
                      />
                      <path
                        d="M4 6V8H6V28a2 2 0 002 2H24a2 2 0 002-2V8h2V6zM8 28V8H24V28zM12 2H20V4H12z"
                      />
                    </svg>
                  </button>
                </div>
              </section>
            </div>
          </div>
          <div
            class="cds--tile _editFieldsCard_734922"
          >
            <section
              class="_editTitleParameters_734922"
            >
              <h1>
                Result Parameters
              </h1>
              <button
                class="cds--btn cds--btn--md cds--btn--ghost"
                style="width: 13.5rem;"
                tabindex="0"
                type="button"
              >
                Add a Result Parameter
                <svg
                  aria-hidden="true"
                  aria-label="Add result parameter"
                  class="cds--btn__icon"
                  fill="currentColor"
                  focusable="false"
                  height="16"
                  preserveAspectRatio="xMidYMid meet"
                  role="img"
                  viewBox="0 0 32 32"
                  width="16"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M17 15L17 8 15 8 15 15 8 15 8 17 15 17 15 24 17 24 17 17 24 17 24 15z"
                  />
                </svg>
              </button>
            </section>
            <div
              class="_fieldsContainer_734922"
            >
              <div
                class="_noFieldsContainer_734922"
              >
                <p
                  class="_noFieldsTitle_734922"
                >
                  No Result Paramaters (yet)
                </p>
                <p
                  class="_noFieldsText_734922"
                >
                  Result Parameters map to the output of a task. Provide the name and description for the variables that will be output as a results of this task.
                </p>
                <p
                  class="_noFieldsText_734922"
                >
                  Add a result paramater above to get started.
                </p>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="cds--bmrg-modal-portal"
  />
  <div
    class="cds--bmrg-modal-portal"
  />
  <div
    class="cds--bmrg-modal-portal"
  />
  <div
    class="cds--bmrg-modal-portal"
  />
  <div
    class="cds--bmrg-modal-portal"
  />
  <div
    class="cds--bmrg-modal-portal"
  />
  <div
    class="cds--bmrg-modal-portal"
  />
  <div
    class="cds--bmrg-modal-portal"
  />
  <div
    class="cds--bmrg-modal-portal"
  />
  <div
    class="cds--bmrg-modal-portal"
  />
  <div
    aria-atomic="true"
    aria-live="assertive"
    id="rbd-announcement-0"
    style="position: absolute; width: 1px; height: 1px; margin: -1px; border: 0px; padding: 0px; overflow: hidden; clip-path: inset(100%);"
  />
  <div
    id="rbd-hidden-text-0-hidden-text-0"
    style="display: none;"
  >
    
  Press space bar to start a drag.
  When dragging you can use the arrow keys to move the item around and escape to cancel.
  Some screen readers may require you to be in focus mode or to use your pass through key

  </div>
</body>
`;
